integration-tests/tests/chain/challenges.rs

```diff
@@ -0,0 +1,117 @@
+mod tests {
```

Comment:
I would call these tests unit tests. They are solely within the realm of chain and does not depend on other parts of the system. In other words, they could be in the same file as `chain/chain/src/chain.rs` for example and it's just that we chose not to do it.